### PR TITLE
Staging next

### PR DESCRIPTION
**Pull request by vcunat**

### CODE REVIEW

**Comment by FRidh** [comment]
Let's see how far we get with b8f7c6e397c771ef63695b3b608c2e64ae0fb9ca

**Comment by domenkozar** [comment]
How come patchelf bump is not here as it was in staging a few days ago?

**Comment by vcunat** [comment]
That's because I've started [pre-building staging](https://hydra.nixos.org/eval/1592635) slightly before patchelf was pushed, so I used that commit.  Patchelf was merged yesterday (CommitDate: Wed Jun 10 10:13:09 2020 +0200).

**Comment by jonringer** [comment]
> Yes ok I will look at it tomorrow morning. I am thinking of actually reverting the change to 3.8. There are so many applications also depending on `aiohttp` that it would mean many people will end up with both 3.7 and 3.8 on their system. cc @jonringer

We could revert the 3.8 bump in staging and push that change to the `python-unstable` branch if it makes stabilizing staging-next easier. 

I can go through and pin all the applications that use aiohttp to python37. But in general I don't think we should put off the python38 bump much longer, it's been available for many months now.

**Comment by FRidh** [comment]
Potential Python 3 fix for darwin https://github.com/NixOS/nixpkgs/pull/90174.

Let's wait and see with 3.8. I am actually quite confident that we won't get many issues due to aiohttp despite the failing tests.

**Comment by vcunat** [comment]
An interesting error in blender:
```
CMake Error at CMakeLists.txt:1517 (message):
  Missing:
  "/nix/store/n0b076p351ma864q38if4yglsg99hw2s-python3-3.8.3/include/python3.8m/Python.h",


  Set the cache entry 'PYTHON_INCLUDE_DIR' to point to a valid python include
  path.  Containing Python.h for python version "3.8"
```

**Comment by vcunat** [comment]
I see no real blocker so far in this iteration, but for darwin there are no builds yet.

**Comment by FRidh** [comment]
That's good. 

Note for the new iteration I would like to have https://github.com/NixOS/nixpkgs/pull/89723 in. It's only leaf packages that are left and that can be fixed on staging-next and master.

File: pkgs/development/tools/misc/autogen/default.nix [hunk 1]
```diff
@@ -62,9 +61,13 @@ stdenv.mkDerivation rec {
       sed -e "s|$bin/bin|/no-such-autogen-bin-path|" -i $f
       sed -e "s|$lib/lib|/no-such-autogen-lib-path|" -i $f
     done
-  '';
 
-  #doCheck = true; # 2 tests fail because of missing /dev/tty
+    # remove /build/** from RPATHs
+    for f in "$bin"/bin/*; do
+      local nrp="$(patchelf --print-rpath "$f" | sed -E 's@(:|^)/build/[^:]*:@\1@g')"
```
Comment (FRidh) [inline pkgs/development/tools/misc/autogen/default.nix:67]: This is problematic for darwin https://hydra.nixos.org/build/122123361

Comment (vcunat) [reply pkgs/development/tools/misc/autogen/default.nix:67]: Oh, right, I think this isn't the first time I forgot about patchelf on darwin.  I can have a look, in about 12h, probably.

Comment (FRidh) [reply pkgs/development/tools/misc/autogen/default.nix:67]: potential fix https://github.com/NixOS/nixpkgs/pull/90284

**Comment by bhipple** [comment]
Found a simple fix while continuing to troubleshoot towards https://github.com/NixOS/nixpkgs/issues/89746

https://github.com/NixOS/nixpkgs/pull/90357

**Comment by nh2** [comment]
With this merged, `kombu` (and thus e.g. `cura`) will fail with a version dependency error on `amqp` in https://hydra.nixos.org/build/122252840#tabs-details

(I noticed it because i accidentally tried to build master.)

**Comment by FRidh** [comment]
Yes, as announced on Discourse we switched to Python 3.8 as default so there will be quite some regressions.

**Comment by ksevelyar** [comment]
> Yes, as announced on Discourse we switched to Python 3.8 as default so there will be quite some regressions.

Is it possible to override python to 3.7 globally? Both Cura and Octoprint broken now.

**Comment by jonringer** [comment]
I'll look into when I wake up.

but yes, it's possible

**Pull request merged**